### PR TITLE
CD-47: Proxmox General monitoring (variables)

### PR DESCRIPTION
**Pull request by mariemm97**

### CODE REVIEW

File: examples/ssh/proxmox_general.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ *      - DNS EXT : The external DNS (Domain Name System) resolution time in milliseconds (ms).
+ * 
+ **/
+
+var cmdPVEVersion = "pveversion";
+var cmdPVEPerf = "pveperf";
+var sshConfig = {
+    username: D.device.username(),
+    password: D.device.password(),
+    timeout: 120000
```
Comment (GeorgiTsatsev) [inline examples/ssh/proxmox_general.js:26]: 2 minutes timeout for a single SSH call is extremely high?

File: examples/ssh/proxmox_general.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        if (parts.length === 2) {
+            var key = parts[0].trim();
+            var value = parseFloat(parts[1].trim());
+            config[key] = value;
+        }
+        variables = [
+            D.createVariable("version", "Version", data[0]),
+            D.createVariable("cpu_bogomips", 'CPU BOGOMIPS', config["CPU BOGOMIPS"]),
+            D.createVariable("regex_second", 'REGEX/SECOND', config["REGEX/SECOND"]),
+            D.createVariable("hd_size", 'HD SIZE', config["HD SIZE"], "GB"),
```
Comment (GeorgiTsatsev) [inline examples/ssh/proxmox_general.js:83]: Is the data always GB

File: examples/ssh/proxmox_general.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+/**
+  * @remote_procedure
+  * @label Validate Association
+  * @documentation This procedure is used to validate if the SSH command is executed successfully.
+  */
+function validate() {
+    executeCommand(cmdPVEVersion)
+        .then(function (data) {
+            if (data && data[0] && data[1]) {
```
Comment (GeorgiTsatsev) [inline examples/ssh/proxmox_general.js:101]: What does this even validate?  That the returned data has two chars?

File: examples/ssh/proxmox_general.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    var config = {};
+    var lines = data[1].split('\n');
+    lines.forEach(function (line) {
+        var parts = line.split(':');
+        if (parts.length === 2) {
+            var key = parts[0].trim();
+            var value = parseFloat(parts[1].trim());
+            config[key] = value;
+        }
+        variables = [
```
Comment (GeorgiTsatsev) [inline examples/ssh/proxmox_general.js:79]: Why is the variable list creation within the loop?

File: examples/ssh/proxmox_general.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    "version": "1.0.0",
+    "name": "Proxmox VE General",
+    "description": "Monitors general and performance properties of a Virtual Machines Proxmox",
+    "category": "Software Integration",
+    "requirements": {
+        "credentials": true,
+        "sandbox_version": "1.12",
+        "others": [
+            "Proxmox",
+            "VM",
```
Comment (GeorgiTsatsev) [inline examples/ssh/proxmox_general.json:12]: What does VM as a requirement mean? 

what about hypervisor?

File: examples/ssh/proxmox_general.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            D.createVariable("version", "Version", data[0]),
+            D.createVariable("cpu_bogomips", 'CPU BOGOMIPS', config["CPU BOGOMIPS"]),
```
Comment (GeorgiTsatsev) [inline examples/ssh/proxmox_general.js:81]: I would argue the Descriptions are more relevant as the variable Labels (still a bit shortened)

Please change the variable labels to something more descriptive and human readable

File: examples/ssh/proxmox_general.js (hunk 2 above)
Comment (mariemm97) [reply examples/ssh/proxmox_general.js:83]: Yes, 
Proxmox uses GB as the default unit for specifying the size of virtual hard disks

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**